### ISSUE TITLE
MATERIALIZED columns are treated as regular columns when inserting data with MATERIALIZED VIEW

### ISSUE THREAD

**Issue by wawaka** [ClickHouse/ClickHouse#3484]
MATERIALIZED columns are not being materialized when inserting data using MV
```sql
DROP TABLE IF EXISTS source;
CREATE TABLE IF NOT EXISTS source (
    n UInt64
) ENGINE = MergeTree
ORDER BY tuple();


DROP TABLE IF EXISTS destination;
CREATE TABLE IF NOT EXISTS destination (
    n UInt64,
    s String MATERIALIZED toString(n)
)
ENGINE = MergeTree
ORDER BY tuple();


DROP TABLE IF EXISTS trigger;
CREATE MATERIALIZED VIEW IF NOT EXISTS trigger
TO destination
AS SELECT * FROM source;


INSERT INTO source VALUES (1);
```

```
INSERT INTO source VALUES

Received exception from server (version 18.14.9):
Code: 10. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: Expected column s: while pushing to view test.trigger. 
```

instead, they are treated as regular columns allowing to set any arbitrary value
```sql
DROP TABLE IF EXISTS source;
CREATE TABLE IF NOT EXISTS source (
    n UInt64
) ENGINE = MergeTree
ORDER BY tuple();


DROP TABLE IF EXISTS destination;
CREATE TABLE IF NOT EXISTS destination (
    n UInt64,
    s String MATERIALIZED toString(n)
)
ENGINE = MergeTree
ORDER BY tuple();


DROP TABLE IF EXISTS trigger;
CREATE MATERIALIZED VIEW IF NOT EXISTS trigger
TO destination
AS SELECT n, toString(n*n) as s FROM source;


INSERT INTO source SELECT * FROM numbers(10);

SELECT n, s FROM destination;
```
```
SELECT 
    n, 
    s
FROM destination 

┌─n─┬─s──┐
│ 0 │ 0  │
│ 1 │ 1  │
│ 2 │ 4  │
│ 3 │ 9  │
│ 4 │ 16 │
│ 5 │ 25 │
│ 6 │ 36 │
│ 7 │ 49 │
│ 8 │ 64 │
│ 9 │ 81 │
└───┴────┘
```

**Comment by v1rusw0rm** [comment]
I tried it on version 19.17.2 and it seems working. Is this issue not closed because it's not tested enough?

**Comment by ZzEeKkAa** [comment]
Worked for me as well

**Comment by den-crane** [comment]
It was implemented in 19.16.
Now MV column set can differ from destination column set.
